### PR TITLE
[HOPSWORKS-759] selenium integration testing

### PR DESCRIPTION
**Pull request by ErmiasG**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [ ] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by ErmiasG** [comment]
logicalclocks/hops-testing#308

File: hopsworks-IT/src/test/java/io/hops/hopsworks/WebDriverFactory.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
+package io.hops.hopsworks;
```
Comment (SirOibaf) [inline hopsworks-IT/src/test/java/io/hops/hopsworks/WebDriverFactory.java]: Missing license

File: hopsworks-IT/pom.xml [hunk 2]
```diff
@@ -0,0 +1,375 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~
+~ This file is part of Hopsworks
+~ Copyright (C) 2018, Logical Clocks AB. All rights reserved
```
Comment (SirOibaf) [inline hopsworks-IT/pom.xml]: 2019

File: hopsworks-IT/src/test/java/io/hops/hopsworks/WebDriverFactory.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+      }
+    }
+
+    if (driver == null) {
+      File chromeDriver = new File(downLoadsDir + CHROMEDRIVER);
+      if (chromeDriver.exists()) {
+        System.setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY, downLoadsDir + CHROMEDRIVER);
+        ChromeOptions chromeOptions = new ChromeOptions();
+        if ("true".equals(System.getenv(BROWSER_UI_ENV))) {
+          chromeOptions.addArguments("--headless");
```
Comment (SirOibaf) [inline hopsworks-IT/src/test/java/io/hops/hopsworks/WebDriverFactory.java]:
```suggestion
          chromeOptions.addArguments("--headless");
          chromeOptions.addArguments("--no-sandbox");
          chromeOptions.addArguments("--disable-dev-shm-usage");
```

Comment (SirOibaf) [reply hopsworks-IT/src/test/java/io/hops/hopsworks/WebDriverFactory.java]: `--no-sandbox` and `--disable-dev-shm-usage` are required to run the tests on jenkins

File: hopsworks-IT/pom.xml [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+      <version>${selenium.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-lang3</artifactId>
+      <version>${commons.lang3.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.rauschig</groupId>
```
Comment (SirOibaf) [inline hopsworks-IT/pom.xml]: Can you add this dependency to the list of dependencies file?

File: hopsworks-IT/pom.xml [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+    <dependency>
+      <groupId>io.hops.hopsworks</groupId>
+      <artifactId>hopsworks-web</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+      <type>war</type>
+    </dependency>
+    <dependency>
+      <groupId>org.seleniumhq.selenium</groupId>
+      <artifactId>selenium-java</artifactId>
```
Comment (SirOibaf) [inline hopsworks-IT/pom.xml]: Can you add this dependency to the list of dependencies file?

File: hopsworks-IT/src/test/java/io/hops/hopsworks/WebDriverFactory.java [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+      chromeVersionCmd = "/Applications/Google\\ Chrome.app/Contents/MacOS/Google\\ Chrome --version";
+    }
+    int firefoxVersion = getVersion(firefoxVersionCmd);
+    int chromeVersion = getVersion(chromeVersionCmd);
+
+    if (chromeVersion >= SUPPORTED_CHROME_VERSION) {
+      File chromeDriver = new File(downloadPath + CHROMEDRIVER);
+      File chromeDriverZip = new File(downloadPath + CHROMEDRIVER + ".zip");
+      downloadDriver(chromeDriver, chromeDriverZip, new File(downloadPath), chromeDriverUrl);
+    } else if (firefoxVersion >= SUPPORTED_FIREFOX_VERSION) {
```
Comment (SirOibaf) [inline hopsworks-IT/src/test/java/io/hops/hopsworks/WebDriverFactory.java]:
```suggestion
    } 
    if (firefoxVersion >= SUPPORTED_FIREFOX_VERSION) {
```

**Comment by tkakantousis** [comment]
@ErmiasG @SirOibaf can this be merged?

**Pull request merged**